src/editors/containers/ProblemEditor/data/OLXParser.js

```diff
@@ -0,0 +1,382 @@
+    }
+
+    data = {
+      answers,
+    };
+
+    return data;
+  }
+
+  parseResponseObject(numericalresponse, answerOffset) {
+    let answerFeedback = '';
+    const answers = [];
+    let responseParam = {};
+    const feedback = this.getFeedback(numericalresponse);
+    if (_.has(numericalresponse, 'responseparam')) {
+      const type = _.get(numericalresponse, 'responseparam.@_type');
+      const defaultValue = _.get(numericalresponse, 'responseparam.@_default');
+      responseParam = {
+        [type]: defaultValue,
+      };
+    }
+    answers.push({
+      id: indexToLetterMap[answers.length + answerOffset],
+      title: numericalresponse['@_answer'],
+      correct: true,
+      feedback,
+      ...responseParam,
+    });
+
+    // Parsing additional_answer for numerical response.
+    const additionalAnswer = _.get(numericalresponse, 'additional_answer', []);
+    if (_.isArray(additionalAnswer)) {
+      additionalAnswer.forEach((newAnswer) => {
+        answerFeedback = this.getFeedback(newAnswer);
+        answers.push({
+          id: indexToLetterMap[answers.length + answerOffset],
+          title: newAnswer['@_answer'],
+          correct: true,
+          feedback: answerFeedback,
+        });
+      });
+    } else {
+      answerFeedback = this.getFeedback(additionalAnswer);
+      answers.push({
+        id: indexToLetterMap[answers.length + answerOffset],
+        title: additionalAnswer['@_answer'],
+        correct: true,
+        feedback: answerFeedback,
+      });
+    }
+    return answers;
+  }
+
+  parseQuestions(problemType) {
+    const builder = new XMLBuilder();
+    const { problem } = this.parsedOLX;
+    const problemObject = _.get(problem, problemType);
+    let questionObject = {};
+    const tagMap = {
+      label: 'bold',
+      description: 'em',
+    };
+
+    /* Only numerical response has different ways to generate OLX, test with
+      numericInputWithFeedbackAndHintsOLXException and numericInputWithFeedbackAndHintsOLX
+      shows the different ways the olx can be generated.
+    */
+    if (_.isArray(problemObject)) {
+      questionObject = _.omitBy(problemObject[0], (value, key) => _.includes(nonQuestionKeys, key));
+    } else {
+      questionObject = _.omitBy(problemObject, (value, key) => _.includes(nonQuestionKeys, key));
+    }
+    // Check if problem tag itself will have question and descriptions.
+    if (_.isEmpty(questionObject)) {
+      questionObject = _.omitBy(problem, (value, key) => _.includes(nonQuestionKeys, key));
+    }
+    const serializedQuestion = _.mapKeys(questionObject, (value, key) => _.get(tagMap, key, key));
+
+    const questionString = builder.build(serializedQuestion);
+    return questionString;
+  }
+
+  getHints() {
+    const { problem } = this.parsedOLX;
+    const hintsObject = [];
+    if (_.has(problem, 'demandhint')) {
+      const { demandhint } = problem;
+      if (_.has(demandhint, 'hint')) {
+        const { hint } = demandhint;
```

Comment:
Nit: We can simlify this:
```suggestion
    if (_.has(problem, 'demandhint.hint')) {
      const hint = _.get(problem, 'demandhint.hint')
```